### ISSUE TITLE
What's the 'right' notion of equality between functions?

### ISSUE THREAD

**Issue by jamesmckinna** [agda/agda-stdlib#2400]
We have [the pointwise `PropositionalEquality` definition `_≗_`](https://github.com/agda/agda-stdlib/blob/c719430e5f27a2df47a18df8781cb5824ba5b601/src/Relation/Binary/PropositionalEquality/Core.agda#L35-L42) (moved in #2335 )
```agda
_≗_ {A = A} {B = B} f g = ∀ x → f x ≡ g x
```
and [the `Function` definition `_≈_`](https://github.com/agda/agda-stdlib/blob/c719430e5f27a2df47a18df8781cb5824ba5b601/src/Function/Relation/Binary/Setoid/Equality.agda#L25-L27) (since #2240 and its antecedent issues back to #1467 )
```agda
_≈_ : (f g : Func From To) → Set (a₁ ⊔ b₂)
f ≈ g = {x : From.Carrier} → f ⟨$⟩ x To.≈ g ⟨$⟩ x
```
and... we are (once again!) inconsistent as to implicit/explicit quantification.

@JacquesCarette has argued for the implicit version; @Taneb 's recent [topic thread on Zulip](https://agda.zulipchat.com/#narrow/stream/263194-categories/topic/Should.20Sets.20use.20.E2.89.97.3F) suggests that this is not always the right choice.

Maybe there's room for both, but it doesn't feel quite right to me... or else, the distinction should be documented.

**Comment by JacquesCarette** [comment]
I agree that it feels inconsistent. Normally, I'd go for implicit when practice shows that it can often be inferred - but it turns out that practice is very mixed on that front! In those cases, I think going explicit is the better choice.

**Comment by MatthewDaggitt** [comment]
I definitely vote for explicit in this case. I often run into places where it can't be inferred.

**Comment by JacquesCarette** [comment]
If we make it "normal style" to use ` _` (i.e. space-underscore) when it can be, then explicit is pretty light-weight.

**Comment by jamesmckinna** [comment]
So... do we *break* the change introduced in #2240 (ahead of v2.1?) and standardise on explicit, moreover (DRY) by delegation of the definition of `_≈_` above to one in terms of `_≗_`?

**Comment by JacquesCarette** [comment]
Since this is indeed 'intra' v2.1, I think this is the way to go.

**Comment by jamesmckinna** [comment]
Closed by #2429

**Comment by jamesmckinna** [comment]
Reopening to reconsider #2381 (and possibly others?), which had previously followed Equality.Setoid in having implicit quantification... and also are 'intra'v2.1 PRs

Some detective work required to track them all (maybe it is just this one?) down, but in the meantime, this is perhaps a wider library design issue that we should discuss?
